context/src/main/java/io/micronaut/runtime/converters/time/TimeConverterRegistrar.java

```diff
@@ -209,8 +209,8 @@ public void register(MutableConversionService conversionService) {
         addTemporalToDateConverter(conversionService, ZonedDateTime.class, ZonedDateTime::toInstant);
         // these two are a bit icky, but required for yaml parsing compatibility
         // TODO Micronaut 4 Consider deletion
-        addTemporalToDateConverter(conversionService, LocalDate.class, ld -> ld.atTime(0, 0).toInstant(ZoneOffset.UTC));
```

Comment:
This line is converting `LocaleDate` to `Date` using `UTC` and micronaut-data is using default timezone when converting back so we are getting date diff issues. I believe this should use default timezone as well. 
There is also comment about removal of this conversion in Micronaut4. Can it be removed? If yes, then we can add converter needed in micronaut-data using default timezone and the issue would be fixed.

Reply:
@yawkat Can we remove these conversions above, or change it to not use `UTC` but default timezone? It is causing issue in micronat-data explained in the description of this PR.

Reply:
I disagree that this should be default TZ. Yes they should be removed, but using system TZ is worse.

Why does data need to convert these at all? java.sql.Date has actual well-defined conversions from and to LocalDate

Reply:
micronaut-data registers various conversions for dates and is always using timezones.
With this converter defined in micronaut-core, even if we register this in micronaut-data
```
conversionService.addConverter(java.sql.Date.class, LocalDate.class, date -> date.toLocalDate());
```
we will get wrong date when converted back.

I removed these converters from micronaut-core, hopefully nothing will break.